RecoLocalCalo/EcalRecAlgos/src/PulseChiSqSNNLS.cc

```diff
@@ -404,6 +420,28 @@ bool PulseChiSqSNNLS::NNLS() {
   
 }
 
+void PulseChiSqSNNLS::NNLSUnconstrainParameter(Index idxp) {
+  
+  aTamat.col(_nP).swap(aTamat.col(idxp));
+  aTamat.row(_nP).swap(aTamat.row(idxp));
+  _pulsemat.col(_nP).swap(_pulsemat.col(idxp));
+  std::swap(aTbvec.coeffRef(_nP),aTbvec.coeffRef(idxp));
+  std::swap(_ampvec.coeffRef(_nP),_ampvec.coeffRef(idxp));
+  std::swap(_bxs.coeffRef(_nP),_bxs.coeffRef(idxp));
+  ++_nP;
```

Comment:
there was also ```      wvec.tail(nActive) = updatework.tail(nActive);  ``` in what's replaced by NNLSUnconstrainParameter call.
Was that essential?